### PR TITLE
Support querying CSV files without providing the schema

### LINKED ISSUE #888: [DataFusion CLI] Support querying CSV files without providing the schema
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
I would like to be able to query CSV files without having to provide the schema.

**Describe the solution you'd like**
The `CREATE EXTERNAL` table command should not require columns to be provided for CSV files. There should be an option for specifying whether the CSV file has a header row with column names.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by xudong963**
# Which issue does this PR close?

Closes #888 

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by xudong963** [comment]
@houqp @alamb PTAL, thanks!

File: datafusion/src/physical_plan/csv.rs [hunk 1]
```diff
@@ -263,12 +264,20 @@ impl CsvExec {
...
         options: &CsvReadOptions,
     ) -> Result<Schema> {
-        Ok(csv::infer_schema_from_files(
-            filenames,
-            options.delimiter,
-            Some(options.schema_infer_max_records),
-            options.has_header,
-        )?)
+        let mut schemas = Vec::new();
+        for filename in filenames {
```
Comment (houqp) [inline datafusion/src/physical_plan/csv.rs:268]: what's the reason for reimplementing `csv::infer_schema_from_files` here?

Comment (xudong963) [reply datafusion/src/physical_plan/csv.rs:268]: Good sight! I deep into `csv::infer_schema_from_files` and find them nearly do the same thing.

File: datafusion/tests/user_defined_plan.rs [hunk 2]
```diff
@@ -109,6 +109,22 @@ async fn setup_table(mut ctx: ExecutionContext) -> Result<ExecutionContext> {
     Ok(ctx)
 }
 
+async fn setup_table_without_columns(
```
Comment (houqp) [inline datafusion/tests/user_defined_plan.rs:112]: i think without schema would have been more accurate here. same for the other tests.

**Review by houqp — APPROVED** [review]
LGTM, thanks @xudong963 !

File: datafusion/tests/user_defined_plan.rs (hunk 2 above)
Comment (xudong963) [reply datafusion/tests/user_defined_plan.rs]: good idea! fixed

**Review by houqp — APPROVED** [review]

File: datafusion/src/sql/planner.rs [hunk 3]
```diff
@@ -196,13 +196,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
 
         // semantic checks
         match *file_type {
-            FileType::CSV => {
```
Comment (alamb) [inline datafusion/src/sql/planner.rs:199]: ❤️

**Review by alamb — APPROVED** [review]
This is great @xudong963  -- wonderful to see that we get new features by effectively deleting code 🏅 

I also tested this out locally and it worked great:

```shell
cargo run -p datafusion-cli
```

And then ran
```sql
> CREATE EXTERNAL TABLE customer STORED AS CSV LOCATION '/Users/alamb/Software/arrow-datafusion/datafusion/tests/customer.csv';
0 rows in set. Query took 0.013 seconds.
> select * from customer;

+----------+----------+
| column_1 | column_2 |
+----------+----------+
| andrew   | 100      |
| jorge    | 200      |
| andy     | 150      |
| paul     | 300      |
+----------+----------+
4 rows in set. Query took 0.013 seconds.
```

Very cool!!! Thank you @xudong963 !

**Pull request merged**